File: решения/cs/Classes/task9713/Interval.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Classes.task9713
{
    public class Interval
    {
        public int a;
        public int b;
        
        public bool Intersects(Interval i)
        {
            return !(b <= i.a || i.b <= a);
        }

        public override string ToString()
        {
            if(a > b)
            {
                return "[Некорректный интервал " + a + ";" + b + "]";
            }
            return "[" + a + ";" + b + "]";
        }
    }
}

File: решения/java/src/Lab14/solution/task9713/Interval.java
package Lab14.solution.task9713;

/**
 * @author akryukov
 *         15.08.2017
 */
public class Interval {
    public int a;
    public int b;

    public boolean intersects(Interval i)
    {
        return !(b <= i.a || i.b <= a);
    }

    @Override
    public String toString()
    {
        if(a > b)
        {
            return "[Некорректный интервал " + a + ";" + b + "]";
        }
        return "[" + a + ";" + b + "]";
    }
}
